simple_certmanager/models.py

```diff
@@ -43,6 +47,20 @@ class Certificate(DeleteFileFieldFilesMixin, models.Model):
         upload_to="ssl_certs_keys/%Y/%m/%d",
         validators=[PrivateKeyValidator()],
     )
+    csr = PrivateMediaFileField(
+        _("certificate signing request"),
+        help_text=_("The content of the certificate signing request"),
+        upload_to="ssl_certs_keys/%Y/%m/%d",
+        blank=True,
+        validators=[CertificateSigningRequestValidator()],
+    )
```

Comment:
I don't think this field should exist on the `Certificate` model - it's only relevant in the request phase but not needed for runtime at all.

Instead, I'd rather add a `models.TextField` on the `CertificateSigningRequest` model where the content of the CSR is stored, that can then easily be copy-pasted for the actual request (and you can add a convenience "download" link in the admin to generate a file with these contents).

Once the certificate itself is added, there's no need to keep the CSR around anymore, so let's just not clutter the file system with them.

Reply:
If compute allows it, CSR could even just be generated on request and not stored at all.